metrics.coffee

```diff
@@ -1,28 +1,56 @@
 StatsD = require('lynx')
 statsd = new StatsD(process.env["STATSD_HOST"] or "localhost", 8125, {on_error:->})
 
+prom = require('prom-client')
+Register = require('prom-client').register
+collectDefaultMetrics = prom.collectDefaultMetrics
+
 name = "unknown"
 hostname = require('os').hostname()
 
 buildKey = (key)-> "#{name}.#{hostname}.#{key}"
 buildGlobalKey = (key)-> "#{name}.global.#{key}"
 
+counters = {}
+gauges = {}
+
 destructors = []
 
 require "./uv_threadpool_size"
 
 module.exports = Metrics =
 	initialize: (_name) ->
 		name = _name
+		collectDefaultMetrics({ timeout: 5000, prefix: name })
 
 	registerDestructor: (func) ->
 		destructors.push func
 
+	injectMetricsRoute: (app) ->
+		app.get('/metrics', (req, res) -> 
```

Comment:
Yes - it's designed to pull metrics, rather than have them pushed. There is a gateway you can run to receive pushed metrics, but that's really intended for transitory things like batch jobs which you can't scrape from.